### PR TITLE
fix: people in budget filter bug default avatar #P23-398 

### PR DESCRIPTION
**Pull request by wyola**
## People in budget - filter bug and default avatar

image

Although I could not reproduce this error, I added some precautions to protect against it. Checking BE's documentation, it looks like `budgetUsers` could be `null` ⬇️ , although I don't know how since there is always at least the budget's owner.

image

If you still see any risks here, please let me know!

I added also the patch for the broken avatar, just in case (thank you @PawelPazderski for generating new svg for me!)
I know the patch is copied for the third time in the project, but idk if we should be extracting it at this stage, so close to demo. I can do it if you wish though.

image

### CODE REVIEW

File: apps/web/app/(navigation)/budgets/[id]/BudgetBasinInformation/PeopleInBudget.tsx [hunk 1]
```diff
@@ -44,6 +45,15 @@ const StyledUser = styled.div`
...
 
+const isAvatarValid = (text: string) => {
+  const schemaPath = z.string().startsWith("/avatars/");
+  const schemaUrl = z.string().url();
+
+  return (
+    schemaPath.safeParse(text).success || schemaUrl.safeParse(text).success
+  );
+};
+
```
Comment (bdeluga) [inline apps/web/app/(navigation)/budgets/[id]/BudgetBasinInformation/PeopleInBudget.tsx:56]: This could be moved into `/lib/validations/`

**Review by bdeluga — APPROVED** [review]
Nice fix 🚀

File: apps/web/app/(navigation)/budgets/[id]/BudgetBasinInformation/PeopleInBudget.tsx (hunk 1 above)
Comment (wyola) [reply apps/web/app/(navigation)/budgets/[id]/BudgetBasinInformation/PeopleInBudget.tsx:56]: @bdeluga I know, but like I wrote, I am not sure if I want to touch it right now xD let's see what the leaders will say!

Comment (monika-klokosinska-intive) [reply apps/web/app/(navigation)/budgets/[id]/BudgetBasinInformation/PeopleInBudget.tsx:56]: @wyola seems like a low risk change - please move it :).

**Review by aniasgit — APPROVED** [review]

**Review by Vectari — APPROVED** [review]
🚀

**Review by AzulloCyll — APPROVED** [review]

**Review by zuzannapaw — APPROVED** [review]
🎸

File: apps/web/app/(navigation)/budgets/[id]/BudgetBasinInformation/PeopleInBudget.tsx (hunk 1 above)
Comment (wyola) [reply apps/web/app/(navigation)/budgets/[id]/BudgetBasinInformation/PeopleInBudget.tsx]: @monika-klokosinska-intive moved

**Pull request merged**